### PR TITLE
Use common favicon fetching/default for all favicons

### LINKED ISSUE #311: Update favicon fallback icon logic
Since we ported the Favicon related stuff, the logic for it got updated on 1.6. This caused some differences in business logic and also in core data models.

We need to update the fallback icon code. I think this PR https://github.com/brave/browser-ios/pull/1644 was the last one where it was touched on 1.6.

### PR DESCRIPTION
**Pull request by kylehickinson**
Switches to using FF's colouring + default favicon loading

Fixes #311 

## Pull Request Checklist

- [ ] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`
- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

### CODE REVIEW

**Comment by kylehickinson** [comment]
Currently still waiting on updated icons for preset favourites, but these can be added when they are available

File: Client/Frontend/Browser/HomePanel/favorites/FavoriteCell.swift [hunk 1]
```diff
@@ -159,7 +130,7 @@ class FavoriteCell: UICollectionViewCell {
     override func prepareForReuse() {
         super.prepareForReuse()
         editButton.isHidden = true
-        showBorder(false)
+//        showBorder(false)
```
Comment (iccub) [inline Client/Frontend/Browser/HomePanel/favorites/FavoriteCell.swift]: Remove

File: Client/Frontend/Menu/HistoryViewController.swift [hunk 2]
```diff
@@ -117,63 +117,12 @@ class HistoryViewController: SiteTableViewController {
     
     cell.imageView?.contentMode = .scaleAspectFit
     cell.imageView?.image = FaviconFetcher.defaultFavicon
+    cell.imageView?.layer.borderColor = UIColor(white: 0, alpha: 0.2).cgColor
```
Comment (iccub) [inline Client/Frontend/Menu/HistoryViewController.swift]: I would extract `UIColor(white: 0, alpha: 0.2)` somewhere, we use it in few places

**Review by iccub — COMMENTED** [review]
Good job, the code looks more simple now.

How hard would it be to show border only for favicons/tiles with light backgrounds?
I think we did it in 1.6.

So for example on the attached screenshot only wikipedia, amazon and coinmarketcap would have borders. We can consult it with design team first.

zrzut ekranu 2018-11-9 o 11 56 53

**Review by jhreis — APPROVED** [review]

**Comment by kylehickinson** [comment]
> How hard would it be to show border only for favicons/tiles with light backgrounds?
I think we did it in 1.6.

Probably not hard :) Though, I don't actually mind the visibility of it for all icons.

**Pull request merged**